lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1107,14 +1107,9 @@ WasmBytecodeGenerator::EmitReturnExpr(EmitInfo *lastStmtExprInfo)
         }
 
         m_writer.Conv(retOp, 0, retExprInfo.location);
-        if (!lastStmtExprInfo)
-        {
-            ReleaseLocation(&retExprInfo);
-        }
     }
     m_writer.AsmBr(m_funcInfo->GetExitLabel());
-
-    return EmitInfo();
+    return retExprInfo;
```

Comment:
I confirm: in both the spec and sexpr-wasm, return does yield a value via desugaring into a value-yielding form of br. This seems to be a discrepancy between the design docs, specs, and implementation. I posted an issue on spec https://github.com/WebAssembly/spec/issues/272 to get some clarity.